lib/plek.rb

```diff
@@ -124,48 +116,37 @@ def website_root
     env_var_or_dev_fallback("GOVUK_WEBSITE_ROOT") { find("www") }
   end
 
-  # Find the base URL for assets.
-  #
-  # @return [URI::HTTPS,URI::HTTP,URI::Generic] The assets base URL.
-  def asset_uri
-    URI(asset_root)
-  end
-
-  # Find the base URL for the public website frontend.
-  #
-  # @return [URI::HTTPS,URI::HTTP,URI::Generic] The website base URL.
-  def website_uri
-    URI(website_root)
-  end
-
-  # TODO: clean up all references to these and then remove them.
   class << self
-    # This alias allows calls to be made in the old style:
-    #     Plek.current.find('foo')
-    # as well as the new style:
-    #     Plek.new.find('foo')
-    alias_method :current, :new
-
-    # Convenience wrapper.  The same as calling +Plek.new.find+.
-    # @see #find
-    def find(*args)
-      new.find(*args)
-    end
-
-    # Convenience wrapper.  The same as calling +Plek.new.find_uri+.
-    # @see #find_uri
-    def find_uri(*args)
-      new.find_uri(*args)
-    end
+    extend Forwardable
+
+    # @!method find
+    #   Convenience wrapper. The same as calling +Plek.new.find+.
+    #   @see #find
+    #   @return [String]
+    # @!method external_url_for
+    #   Convenience wrapper. The same as calling +Plek.new.external_url_for+.
+    #   @see #external_url_for
+    #   @return [String]
+    # @!method asset_root
+    #   Convenience wrapper. The same as calling +Plek.new.asset_root+.
+    #   @see #asset_root
+    #   @return [String]
+    # @!method website_root
+    #   Convenience wrapper. The same as calling +Plek.new.website_root+.
+    #   @see #website_root
+    #   @return [String]
+    def_delegators :new, :find, :external_url_for, :asset_root, :website_root
   end
 
 private
 
   attr_reader :host_prefix, :unprefixable_hosts, :use_http_for_single_label_domains
 
-  # TODO: clean up call sites throughout alphagov and then delete clean_name.
-  def clean_name(service)
-    service.to_s.downcase.strip.gsub(/[^a-z.-]+/, "")
+  def valid_service_name(name)
+    service_name = name.to_s
+    return service_name if service_name.match(/\A[a-z.-]+\Z/)
```

Comment:
- Consider using `match?` for clarity and to avoid unnecessary allocations.
- We don't currently have any services with digits in their names, but it's reasonable to expect that someone might want to someday (`-v2` of a microservice, or abbreviations such as `e2e`, `p2p`, `b2c` etc.)
- Use the much more common `^` and `$` anchors (which are semantically equivalent in this case since it's not a multi-line regex).
```suggestion
    return service_name if service_name.match?(/^[a-z0-9.-]+$/)
```

Reply:
- Good shout on `match?` - I'd intended to use that
- Yeah numbers sounds ok, I've added that in. I talked myself down from an earlier more complex version which required alpha character as first character and ended up just falling back to what was already defined. But I think number sounds reasonable so have added that in.
- It's not actually equivalent ^ and $ just match until a new line character. I've added a test that fails with ^ and $ but not \A and \z. I'm sure the Ruby quirks of this are well documented someone but my Google skills are failing me.

Reply:
Oh interestingly this approach has a bug: it doesn't catch a newline at the end of the string (neither with `$` nor with `\Z`). Which itself seems to be a design bug in Ruby's regexp library as far as I can tell 🧐 

```
irb(main):001:0> "a\n".match?(/a$/)
=> true
irb(main):002:0> "a\n".match?(/a\Z/)
=> true
irb(main):003:0> "a\n".match?(/a$/m)
=> true
irb(main):004:0> "a\n".match?(/a\Z/m)
=> true
```

We probably don't need to care about this case though, since we're just trying to warn in case of likely erroneous inputs. If a stray line ending finds its way into a hostname, it'll get caught somewhere downstream anyway.

Reply:
Sorry that was my mistake too. I fixed it in my most recent push: https://github.com/alphagov/plek/compare/e364af62cd552e1ec071c3ad59d4232c8c9bb221..efa7df54372e025f84a5e6b2f7c27d2208ec3a33

Using \z rather than \Z will catch the rogue new line. I figured it was too niche to add a test case for.

Reply:
Ohh we need `\z` (lowercase `z`) in Ruby. Just to be different from literally every other Perl-inspired RE implementation 🤦😂

Reply:
Ah sorry GitHub was being slow there, reloaded the page and saw your replies. LGTM.